src/derivative_operators/BC_operators.jl

```diff
@@ -0,0 +1,160 @@
+abstract type AbstractBC{T} end
+
+# robin, general, and in general neumann BCs are all affine opeartors, meaning that they take the form Qx = Qax + Qb. neumann0 is not however; is a specialization needed?
+abstract type AffineBC{T,V} <: AbstractBC{T} end
+
+struct DirichletBC{T} <: AbstractBC{T}
```

Comment:
Dirichlet is also affine if `l` and `r` aren't 0.

Reply:
fixed comment